pandas/tests/test_index.py

```diff
@@ -2959,6 +2959,29 @@ def test_nat(self):
         self.assertIs(DatetimeIndex([np.nan])[0], pd.NaT)
 
 
+    def test_contains(self):
+
+        i = self.create_index()
```

Comment:
instead of copy-pasting :)

You can make a new mixin-class like ``DatetimeLike`` and just include it in``DatetimeIndex/PeriodIndex``